src/rules/order.js

```diff
@@ -890,6 +890,14 @@ module.exports = {
           },
         },
         additionalProperties: false,
+        dependencies: {
+          'newlines-between-types': {
+            properties: {
+              sortTypesGroup: { enum: [true] },
+            },
+            required: ['sortTypesGroup'],
+          },
+        },
```

Comment:
@ljharb I came up with this after reading the json schema docs. I'm not sure if ESLint's rule tester can test for schema validation errors? I did some ad-hoc testing and it appears functional

Reply:
This got me thinking: should a dependency also be added between `sortTypesGroup` and `groups` (ensuring `groups` contains `'types'`)? Should we add a dependency between `consolidateIslands` and `newlines-between`/`newlines-between-types` (ensuring at least one is set to `'always-and-inside-groups'`)?

Reply:
Generally, anything that doesn’t make sense should be prohibited by the schema.

Reply:
Just making sure, since the `sortTypesGroup` PR was already accepted :)

I'll open a new PR for that change

Reply:
hmm, i'm confused - what is `dependencies`? i'm not familiar with that field in eslint rule schemas.

Reply:
I just discovered it the other day. It's the Draft-04 version of [`dependentSchemas`](https://json-schema.org/understanding-json-schema/reference/conditionals#dependentSchemas) and was the best way I could find to conditionally require `sortTypesGroups: true` only when `newlines-between-types` was present